### PR TITLE
support if-modified-since specifications

### PR DESCRIPTION
**Pull request by lorenzofox3**
This pull request aims to improve the cache capabilities of the web server using "if-modified-since" specification. 

Note I have not provided unit tests as I was not sure how to check for headers with the spec helpers.
I have however tested manually and added placeholder for the specs. 

PS: I ll be away for couple of weeks

### CODE REVIEW

**Comment by evantahler** [comment]
This is a good start!

- You have updated how the genericServer and webServer deal with files to handle the new lastModified param.  You need to also update the websocket and socket servers.
- for testing, since you are truly testing the HTTP transport, use request.  Here are some examples: https://github.com/evantahler/actionhero/blob/master/test/servers/web.js#L289-L349

**Comment by lorenzofox3** [comment]
great, thanks. just a little head up to let you know I am back from my vacation and hopefully complete the PR this week. Thanks for your patience

**Comment by lorenzofox3** [comment]
Updated with the spec. 

Not sure if the if-modified-since should be implemented with socket and websocket as they don't use the same specifications for the transport protocol ? (not familiar with these protocoles)

File: config/redis.js [hunk 1]
```diff
@@ -6,16 +6,18 @@ exports.default = {
...
     if( process.env.FAKEREDIS === 'false' || process.env.REDIS_HOST !== undefined ){
       redisDetails.package  = 'ioredis';
       // there are many more connection options, including support for cluster and sentinel
       // learn more @ https://github.com/luin/ioredis
-      redisDetails.host     = process.env.REDIS_HOST || '127.0.0.1';
-      redisDetails.port     = process.env.REDIS_PORT || 6379;
-      redisDetails.database = process.env.REDIS_DB   || 0;
       redisDetails.options  = {
         password: (process.env.REDIS_PASS || null),
       };
```
Comment (evantahler) [inline config/redis.js]: Please rebase this file.  I assume you didn't mean to change it.

File: initializers/connections.js [hunk 2]
```diff
@@ -37,7 +37,11 @@ module.exports = {
...
         if(method && args){
-          connection[method].apply(connection, args);
+          if(method === 'sendMessage' || method === 'sendFile'){
+            connection[method](args);
+          }else{
+            connection[method].apply(connection, args);
+          }          
         }
         if(typeof callback === 'function'){
           process.nextTick(function(){
```
Comment (evantahler) [inline initializers/connections.js]: same here

**Comment by evantahler** [comment]
I think you had a problem with your merge/rebase.  You have a lot of files here that shouldn't be included

**Pull request closed without merging**